pkg/controller/machine.go

```diff
@@ -1036,8 +1036,91 @@ func shouldReconcileMachine(machine *v1alpha1.Machine, now time.Time) bool {
 	return true
 }
 
+// // setMachineConfigurations checks if there are any machine-controller configurations on the machine-object, and sets if there are any.
+// func (c *controller) getEffectiveMachineConfigurations(machine *v1alpha1.Machine) error {
+
+// 	if machine.Spec.MachineConfiguration != nil {
+// 		if machine.Spec.MachineConfiguration.MachineDrainTimeout != nil {
+// 			effectiveDrainTimeout := *machine.Spec.MachineConfiguration.MachineDrainTimeout
+// 		}
+
+// 		if machine.Spec.MachineConfiguration.MachineHealthTimeout != nil {
+// 			effectiveHealthTimeout = *machine.Spec.MachineConfiguration.MachineHealthTimeout
+// 		}
+
+// 		if machine.Spec.MachineConfiguration.MachineCreationTimeout != nil {
+// 			effectivet = *machine.Spec.MachineConfiguration.MachineCreationTimeout
+// 		}
+
+// 		if machine.Spec.MachineConfiguration.MaxEvictRetries != nil {
+// 			c.safetyOptions.MaxEvictRetries = *machine.Spec.MachineConfiguration.MaxEvictRetries
+// 		}
+
+// 		if machine.Spec.MachineConfiguration.NodeConditions != nil {
+// 			c.nodeConditions = *machine.Spec.MachineConfiguration.NodeConditions
+// 		}
+// 	}
+
+// 	return nil
+// }
+
 // decodeMachineID is a generic way of decoding the Spec.ProviderID field of node-objects.
 func decodeMachineID(id string) string {
 	splitProviderID := strings.Split(id, "/")
 	return splitProviderID[len(splitProviderID)-1]
 }
+
+// getEffectiveDrainTimeout returns the drainTimeout set on the machine-object, otherwise returns the timeout set using the global-flag.
+func (c *controller) getEffectiveDrainTimeout(machine *v1alpha1.Machine) *metav1.Duration {
+	var effectiveDrainTimeout *metav1.Duration
+	if machine.Spec.MachineConfiguration.MachineDrainTimeout != nil {
+		effectiveDrainTimeout = machine.Spec.MachineConfiguration.MachineDrainTimeout
+	} else {
+		effectiveDrainTimeout = &c.safetyOptions.MachineDrainTimeout
+	}
+	return effectiveDrainTimeout
+}
+
+// getEffectiveMaxEvictRetries returns the maxEvictRetries set on the machine-object, otherwise returns the evict retries set using the global-flag.
+func (c *controller) getEffectiveMaxEvictRetries(machine *v1alpha1.Machine) *int32 {
+	var maxEvictRetries *int32
+	if machine.Spec.MachineConfiguration.MaxEvictRetries != nil {
+		maxEvictRetries = machine.Spec.MachineConfiguration.MaxEvictRetries
+	} else {
+		maxEvictRetries = &c.safetyOptions.MaxEvictRetries
+	}
+	return maxEvictRetries
+}
```

Comment:
```suggestion
func (c *controller) getEffectiveMaxEvictRetries(machine *v1alpha1.Machine) int32 {
	var maxEvictRetries int32
	if machine.Spec.MachineConfiguration.MaxEvictRetries != nil {
		maxEvictRetries = *machine.Spec.MachineConfiguration.MaxEvictRetries
	} else {
		maxEvictRetries = c.safetyOptions.MaxEvictRetries
	}
	return maxEvictRetries
}
```
Change return type to int32 so that it can be directly used without dereferencing?  Only suggesting, feel free to ignore if you feel otherwise.

Reply:
I'd generally be fine, to keep it consistent with similar methods.

Reply:
Cool.